network/common/src/main/java/org/apache/spark/network/util/ByteUnit.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.network.util;
+
+/**
+ * Code based on https://github.com/fabian-barney/Utils
```

Comment:
Sean - I think his primary reason for using ```double```s was to allow handling of larger byte sizes. On second thought though, using long allows us up to 16 Exabytes which should be more than we ever need. 

I'll rewrite it from scratch to avoid the copyright issues and to fix this.